src/Core/Binding/IActiveSolutionBoundTracker.cs

```diff
@@ -35,6 +35,15 @@ public interface IActiveSolutionBoundTracker
         /// </remarks>
         BindingConfiguration CurrentConfiguration { get; }
 
+        /// <summary>
+        /// Raised after the binding has been updated, and before the <see cref="SolutionBindingUpdated"/> event
+        /// </summary>
+        /// <remarks>Most listeners should use the <see cref="SolutionBindingChanged"> event.
+        /// This event is intended for components that need to recalculate/refresh data that
+        /// is then used by listeners that consume the <see cref="SolutionBindingChanged"/> event
+        /// e.g. components that cache solution-level data.</remarks>
+        event EventHandler<ActiveSolutionBindingEventArgs> PreSolutionBindingChanged;
+
```

Comment:
Added two new 'PreXXX' events with documentation.

Calling the events e.g. `BeforeSolutionBindingChanged` doesn't make sense because the solution binding has already changed, so I went with `Pre` as the prefix rather than `Before`.